### PR TITLE
Stability improvements

### PR DESCRIPTION
**Pull request by Segfaultd**
### Introduction

This pull request fixes a few points:

- Fix the docker compose for local development environment
- Bump the used dependencies
- Use DTO for public endpoints params to avoid having weird type conversion being passed to the database
- Handle long-responding endpoints potential timeouts
- Improve chain service readability by revamping comments
- Improve chain service with more correct values being returned in case of error

### Testing

WIP

### TODO before merging:
- Export prod table asset from DB
- Align with FE team for endpoint change

### CODE REVIEW

File: src/services/proposal.service.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    fetch = async (): Promise<ProposalEntity[]> => {
+        return this._repository.find({
+            order: {
+                id: 'DESC',
+            },
+        });
+    };
+
+    createOrUpdateProposal = async (data: Partial<ProposalEntity>) => {
```
Comment (Ricardo-Remy) [inline src/services/proposal.service.ts:24]: Could we use the `findOneAndUpdate` from TypeOrm ?
```
 createOrUpdateProposal = async (data: Partial<ProposalEntity>) => {
        const proposal = await this._repository.findOneAndUpdate({ id: data.id }, { $set: data }, { new: true, upsert: true });
        return proposal;
    };
```

File: src/services/chains/generic.chain.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    getTokenSupply = async (): Promise<number> => {
+        try {
+            const supply = Number((await this.client.getSupply(this.microDenom)).amount) / TEN_EXPONENT_SIX;
+            return supply;
+        } catch (error) {
+            return 0;
+        }
+    };
+
+    getAPY = async (): Promise<number> => {
```
Comment (Ricardo-Remy) [inline src/services/chains/generic.chain.ts:142]: Should we follow the try catch principle as getTokenSupply and return 0 in a try catch ? Or is computeTotalApy returning 0 already if function not executed ?

File: src/services/chains/generic.chain.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        const inflation = Number(await this.client.queryClient.mint.inflation()) / CLIENT_PRECISION;
+        const metrics = await computeTotalApy(this.client, Number(await this.getTokenSupply()), inflation, CLIENT_PRECISION, TEN_EXPONENT_SIX);
+        return apy(metrics.inflation, metrics.communityTaxRate, metrics.stakingRatio);
+    };
+
+    getTotalAllocatedToken = async (): Promise<number> => {
```
Comment (Ricardo-Remy) [inline src/services/chains/generic.chain.ts:148]: Should we follow the try catch principle as getTokenSupply and return 0 in the catch ?

File: src/utils/dfract.ts [hunk 4]
```diff
@@ -5,14 +5,7 @@ export const apy = (inflation: number, communityTaxRate: number, stakingRatio: n
 
 // Helper function to compute total token amount from chains where we can get the info
 export const computeTotalTokenAmount = async (getDecodedAddress: string, client: LumClient, denom: string, applyClientPrecision: number, applyTenExponentSix: number): Promise<number> => {
```
Comment (Ricardo-Remy) [inline src/utils/dfract.ts:7]: `applyClientPrecision` is declared but its value is never read

File: src/services/lum-network.service.ts [hunk 5]
```diff
@@ -212,28 +153,26 @@ export class LumNetworkService {
...
 
-    getAllocatedToken = async () => {
+    getAllocatedToken = async (): Promise<number> => {
         try {
-            // Compute total allocated token for Lum Network
             const token = await computeTotalTokenAmount(LUM_STAKING_ADDRESS, this.client, LumConstants.MicroLumDenom, CLIENT_PRECISION, TEN_EXPONENT_SIX);
-
             return token;
         } catch (error) {
+            console.error(error);
```
[the remark this answers is not in the document]
Comment (Segfaultd) [reply src/services/lum-network.service.ts:172]: Builtin logger won't dump the stacktrace. So depends of the usage. That one was actually unused

File: src/services/proposal.service.ts (hunk 1 above)
Comment (Segfaultd) [reply src/services/proposal.service.ts:24]: Update won't create, not the same behavior

**Pull request merged**